File: spring-boot-cinema/src/main/java/com/luv2code/cinema/service/ReservationService.java
package com.luv2code.cinema.service;

import com.luv2code.cinema.entity.Projection;
import com.luv2code.cinema.entity.Reservation;
import com.luv2code.cinema.entity.Salle;
import com.luv2code.cinema.entity.User;

import java.util.List;

public interface ReservationService {

    Reservation add(Reservation reservation);
    List<Reservation> findAll();
    Reservation save(Reservation reservation);
    List<Reservation> findByUserId(Long id);
    void delete(Long id);


}

File: angular-Cinema-frontend/src/app/services/reservation.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Reservation } from '../models/Reservation';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class ReservationService {

  constructor(private _http: HttpClient) { }

  baseUrl = environment.baseUrl;

  findAll() {
    return this._http.get<Reservation[]>(this.baseUrl + "reservation/find-all");
  }

  delete(id: any) {
    return this._http.delete(this.baseUrl + `reservation/${id}`);
  }
  findByUserId(id: any) {
    return this._http.get<Reservation[]>(this.baseUrl + `reservation/find-by-user-id/${id}`);
  }
  
  annulerRes(id:any){
    return this._http.delete(this.baseUrl + `reservation/${id}`);
  }

  save(reservation: any) {
    return this._http.post(this.baseUrl + "reservation/add", reservation)
  }
}
